api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java

```diff
@@ -229,10 +245,47 @@ public void onInventoryClick(InventoryClickEvent event) {
             final int rawSlot = event.getRawSlot();
             if (rawSlot < 3 || event.getAction().equals(InventoryAction.MOVE_TO_OTHER_INVENTORY)) {
                 event.setCancelled(!interactableSlots.contains(rawSlot));
-                final List<ResponseAction> actions =
+                if (clickHandlerRunning) {
+                    // A click handler is running, don't launch another one
+                    return;
+                }
+
+                final CompletableFuture<List<ResponseAction>> actionsFuture =
                         clickHandler.apply(rawSlot, StateSnapshot.fromAnvilGUI(AnvilGUI.this));
-                for (final ResponseAction action : actions) {
-                    action.accept(AnvilGUI.this, clicker);
+                // Set to running after calling the ClickHandler if the ClickHandler instantly throws an exception
+                clickHandlerRunning = true;
```

Comment:
Why do we deal with setting `clickHandlerRunning` if the future is immediately completed? Why not just move this to the else block?